components/builder-depot/src/server.rs

```diff
@@ -91,7 +88,12 @@ impl TestableBroker {
         let msg_type = &TypeId::of::<M>();
         match self.message_map.get(msg_type) {
             Some(message) => Ok(parse_from_bytes::<R>(message).unwrap()),
-            None => panic!("Unable to find a message of type {:?}", msg_type),
+            None => {
+                match self.error_map.get(msg_type) {
+                    Some(error) => Err(error.clone()),
+                    None => panic!("Unable to find a message of type {:?}", msg_type),
```

Comment:
I actually do want to panic here (but not below). The only case where this would panic is in a test, and in that and that would indicate to the developer that they for got to setup a connection route. In such a case, a stacktrace will help to pin this down.